internal/clients/hvn_route.go

```diff
@@ -28,3 +31,73 @@ func ListHVNRoutes(ctx context.Context, client *Client, hvnID string,
 
 	return listHVNRoutesResponse.Payload.Routes, nil
 }
+
+// DeleteSnapshotByID deletes an HVN route by its ID
+func DeleteHVNRouteByID(ctx context.Context, client *Client, hvnID string,
+	hvnRouteID string, loc *sharedmodels.HashicorpCloudLocationLocation) (*networkmodels.HashicorpCloudNetwork20200907DeleteHVNRouteResponse, error) {
+	deleteHVNRouteParams := network_service.NewDeleteHVNRouteParams()
+
+	deleteHVNRouteParams.Context = ctx
+	deleteHVNRouteParams.ID = hvnRouteID
+	deleteHVNRouteParams.HvnID = hvnID
+	deleteHVNRouteParams.HvnLocationOrganizationID = loc.OrganizationID
+	deleteHVNRouteParams.HvnLocationProjectID = loc.ProjectID
+
+	deleteHVNRouteResponse, err := client.Network.DeleteHVNRoute(deleteHVNRouteParams, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return deleteHVNRouteResponse.Payload, nil
+}
+
+const (
+	// HvnRouteStateCreating is the CREATING state of an HVN route
+	HvnRouteStateCreating = string(networkmodels.HashicorpCloudNetwork20200907HVNRouteStateCREATING)
+
+	// HvnRouteStateActive is the ACTIVE state of an HVN route
+	HvnRouteStateActive = string(networkmodels.HashicorpCloudNetwork20200907HVNRouteStateACTIVE)
+
+	// HvnRouteStatePending is the PENDING state of an HVN route
+	HvnRouteStatePending = string(networkmodels.HashicorpCloudNetwork20200907HVNRouteStatePENDING)
+)
+
+// hvnRouteRefreshState refreshes the state of the HVN route by calling
+// the LIST endpoint
+func hvnRouteRefreshState(ctx context.Context, client *Client, destination string, hvnID string, loc *sharedmodels.HashicorpCloudLocationLocation) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		route, err := ListHVNRoutes(ctx, client, hvnID, destination, "", "", loc)
+		if err != nil {
+			return nil, "", err
+		}
+
+		if len(route) != 1 {
+			return nil, "", fmt.Errorf("Unexpected number of HVN route returned when waiting for route with destination CIDR of %s for HVN (%s) to be Active", destination, hvnID)
```

Comment:
I think error message isn't correct here

Reply:
Previous block calls the LIST endpoint with the destination cidr filter, and should only return 1 single route, therefore this error is here. What do you think is incorrect about it?